api/src/services/graphql/index.ts

```diff
@@ -1688,10 +1658,35 @@ export class GraphQLService {
 					}) ?? [];
 		}
 
+		query.filter = this.replaceFuncs(query.filter);
+
 		validateQuery(query);
 
 		return query;
 	}
+
+	/**
+	 * Replace functions from GraphQL format to Directus-Filter format
+	 */
+	replaceFuncs(filter?: Filter | null): null | undefined | Filter {
+		if (!filter) return filter;
+
+		return replaceFuncDeep(filter);
+
+		function replaceFuncDeep(filter: Record<string, any>) {
+			return transform(filter, (result: Record<string, any>, value, key) => {
+				if (typeof key === 'string' && key.endsWith('_func')) {
```

Comment:
Upon testing, I realised that my above comment wasn't accurate. The function name is nested, and not in the collection name itself, unlike #16195. 🙈  

Patched it by skipping the transformation of fields ending with `_func` if its child key isn't a valid function. 
Would this patch suffice? 🤔 

It should now only be an issue if and only if...

- Querying in GraphQL and filtering with:
  - collection name ending with `_func`
  - nested field name matching one of the following functions

https://github.com/directus/directus/blob/1d08deefbec46f1e2804121a02d221974c189b23/packages/shared/src/constants/fields.ts#L57

Reply:
It should suffice as that should make "false positives" much lower than the previous (existing) logic 👍

On a similar note as before, let's move `functionName` upwards so we can reuse it in the `.includes` then we're good to go 🙌

![](https://user-images.githubusercontent.com/42867097/201252921-e54d1bbf-ea9f-4233-8942-597af4bbef44.png)

Reply:
> that should make "false positives" much lower

Awesome! 😁 

> let's move functionName upwards

I had chose not to move it up as it felt a little odd having the `functionName` outside of the functions block and also due to additional checks required. The following code was added initially, but had decided against it being irrelevant for "non-functions" and for simplicity.

```ts
let functionName = '';
const isFunctionKey =
	key.endsWith('_func') && FUNCTIONS.includes((functionName = Object.keys(value)[0]!) as any);
```

By shifting it up, I believe additional checks are needed for `value`, as it might be `null`, leading to an error thrown by `Object.keys()`. 
Eg: filter value being `"_eq": "$CURRENT_USER"` but user isn't logged in.

The `replaceFuncs()` runs after `sanitizeQuery()` that parses dynamic variables which may return a `null` value.

https://github.com/directus/directus/blob/3ea065f124dbb70b2731989a2aa196a793d8ced5/packages/shared/src/utils/parse-filter.ts#L129-L132  

Thoughts? 🤔

Reply:
Thanks for clarifying! It being the second condition thus it will never get ran outside of `key.endsWith('_func')` was the key! (no pun intended)

The current approach sounds good to me 👍